Release/src/http/client/http_client_winhttp.cpp

```diff
@@ -1220,9 +1254,39 @@ class winhttp_client : public _http_client_communicator
                         break;
                     }
 
+                    auto writebuf = p_request_context->_get_writebuffer();
+
+                    // If we have compressed data it is stored in the local allocation of the p_request_context. We will store the decompressed buffer in the external allocation of the p_request_context.
+                    if (p_request_context->decompressor)
+                    {
+                        web::http::details::compression::data_buffer decompressed = p_request_context->decompressor->decompress(p_request_context->m_body_data.get(), bytesRead);
+
+                        if (p_request_context->decompressor->has_error())
+                        {
+                            p_request_context->report_exception(std::runtime_error("Failed to decompress the response body"));
+                            return;
+                        }
+
+                        // We've decompressed this chunk of the body, need to now store it in the writebuffer.
+                        auto decompressed_size = decompressed.size();
+
+                        if (decompressed_size > 0)
+                        {
+                            auto p = writebuf.alloc(decompressed_size);
+                            p_request_context->allocate_reply_space(p, decompressed_size);
+                            std::memcpy(p_request_context->m_body_data.get(), &decompressed[0], decompressed_size);
```

Comment:
So, I'm not happy with this copy, but the difficulty is that we cannot know the size of the decompressed data without actually decompressing the data, and to allocate in the writebuf we need to know the size. 

Any ideas?